### ISSUE TITLE
SQL Server RETURNING clause emulation for fetching server side computed columns generates incorrect SQL if only computed columns are fetched

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14703]
There's another, similar issue like https://github.com/jOOQ/jOOQ/issues/14479, where the emulation of `RETURNING` produces invalid SQL, with the following preconditions:

- The table contains server side computed columns
- We fetch *only* those columns in `RETURNING`, no others

The generated SQL looks like this:

```sql
declare @result table (
  [cd] ,    -- No data types on these computed columns
  [cs] ,    -- No data types on these computed columns
  [cv] ,    -- No data types on these computed columns
  [id] int
);
insert into [test].[dbo].[t_computed_constant] ([id], [val])
output inserted.[cd], inserted.[cs], inserted.[cv], inserted.[id]
into @result
values (
  1, 
  0
);
merge into @result [r]
using (
  select
    [test].[dbo].[t_computed_constant].[cd] [alias_119969359],
    [test].[dbo].[t_computed_constant].[cs] [alias_119969824],
    [test].[dbo].[t_computed_constant].[cv] [alias_119969917],
    [test].[dbo].[t_computed_constant].[id]
  from [test].[dbo].[t_computed_constant]
) [s]
on [r].[id] = [s].[id]
when matched then update set -- No set clause available
;
select [cd], [cs], [cv]
from @result [r];
```

Like in https://github.com/jOOQ/jOOQ/issues/14479, we shouldn't generate the `MERGE` statement, because if we fetch *only* server side computed columns, then there cannot be any additional trigger generated values, so `MERGE` isn't necessary.

The missing data types are caused by the fact that they are illegal in ordinary DDL (see https://github.com/jOOQ/jOOQ/issues/6492), but in this case, when declaring the `@result` table, we should remove the generator from the data type, thus declaring the data type explicitly.

**Comment by lukaseder** [comment]
After a fix, the generated query looks like this:

```sql
declare @result table (
  [cd] int,
  [cs] int,
  [cv] int,
  [id] int
);
insert into [test].[dbo].[t_computed_constant] ([id], [val])
output inserted.[cd], inserted.[cs], inserted.[cv], inserted.[id]
into @result
values (
  1, 
  0
);
select [cd], [cs], [cv]
from @result [r];
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0 and 3.17.9 (https://github.com/jOOQ/jOOQ/issues/14704)
